contracts/fee/FeeHandler.sol

```diff
@@ -3,46 +3,229 @@
 pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
 import "../data/DataStore.sol";
 import "../role/RoleModule.sol";
 import "../fee/FeeUtils.sol";
+import "../data/Keys.sol";
+import "../v1/IVaultV1.sol";
+import "./IFeedAddress.sol";
 
 // @title FeeHandler
 contract FeeHandler is ReentrancyGuard, RoleModule {
+    using SafeERC20 for IERC20;
+
     DataStore public immutable dataStore;
     EventEmitter public immutable eventEmitter;
+    address public immutable vault;
+    address public immutable gmx;
+    address public immutable wnt;
 
     constructor(
         RoleStore _roleStore,
         DataStore _dataStore,
-        EventEmitter _eventEmitter
+        EventEmitter _eventEmitter,
+        address _vault,
+        address _gmx,
+        address _wnt
     ) RoleModule(_roleStore) {
         dataStore = _dataStore;
         eventEmitter = _eventEmitter;
+        vault = _vault;
+        gmx = _gmx;
+        wnt = _wnt;
     }
 
     // @dev claim fees from the specified markets
     // @param markets the markets to claim fees from
-    // @param tokens the fee tokens to claim
+    // @param feeTokens the fee tokens to claim for (GMX/WNT)
+    // @param swapTokens the provided swap token (GMX/WNT)
     function claimFees(
-        address[] memory markets,
-        address[] memory tokens
+        address[] calldata markets,
```

Comment:
i think we should change this to claimFees for only one market and one buybackToken, the logic seems a bit complicated for an array now

Reply:
Updated to remove the loop in the latest commit